rust/arrow/src/array/mod.rs

```diff
@@ -85,6 +85,7 @@ mod array;
 mod builder;
 mod data;
 mod equal;
+mod union;
```

Comment:
I agree that it makes sense to have submodules under array.

Reply:
Yea, even just to make the tests categorized better.  I'll open a JIRA once this is merged.